client/mysqlslap.c

```diff
@@ -1994,6 +2030,29 @@ pthread_handler_t run_task(void *p)
 
       if (con->limit && queries == con->limit)
         goto end;
+
+      if (unlikely(con->srv_time_target))
+      {
+        gettimeofday(&tmstamp, NULL);
+        /*
+          srv_time set as a ulong value based on the assumption that time diff
+          between two execs never exceed 4293sec (: ((2^32-1)-1000^2)/(1000^2) ).
+        */
+        ulong srv_time= (tmstamp.tv_sec - last_tmstamp.tv_sec) * 1000000 +
+                         tmstamp.tv_usec - last_tmstamp.tv_usec;
+        last_tmstamp= tmstamp;
+        /*
+          in case srv_time_target is rather big, random gain helps queries
+          to be sparsely distributed.
+         */
+        if (con->srv_time_target > srv_time)
```

Comment:
Just curious here. Is increasing or decreasing sleep_tm better then just setting it? Have you seen specific improvements caused by this approach? It took me an extra bit more to double check that you will not overflow it.